packages/aws-cdk-lib/aws-codebuild/lib/fleet.ts

```diff
@@ -35,6 +35,69 @@ export interface FleetProps {
    * made available to projects using this fleet
    */
   readonly environmentType: EnvironmentType;
+
+  /**
+   * The compute configuration of the compute fleet.
+   *
+   * This is only required if `computeType` is set to ATTRIBUTE_BASED.
+   *
+   * @see https://docs.aws.amazon.com/codebuild/latest/userguide/build-env-ref-compute-types.html#environment-reserved-capacity.types
+   *
+   * @default - do not specify compute configuration
+   */
+  readonly computeConfiguration?: ComputeConfiguration;
+}
+
+/**
+ * The compute type of the fleet.
+ */
+export enum MachineType {
+  /**
+   * General purpose compute type.
+   */
+  GENERAL = 'GENERAL',
+  /**
+   * Non-Volatile Memory Express (NVMe) storage optimized compute type.
+   */
+  NVME = 'NVME',
+}
+
+/**
+ * The compute configuration for the fleet.
+ *
+ *  Despite what the CloudFormation schema says, the numeric properties (disk, memory, vCpu) are not optional.
+ *  An `undefined` value will cause the CloudFormation deployment to fail, e.g.
+ *  > Cannot invoke "java.lang.Integer.intValue()" because the return value of "software.amazon.codebuild.fleet.ComputeConfiguration.getMemory()" is null
+ * Therefore, these properties default value is set to 0.
+ */
+export interface ComputeConfiguration {
+  /**
+   * The amount of disk space of the instance type included in your fleet.
+   *
+   * @default 0
```

Comment:
I don't know if it's necessary to tell the user we set the default value to 0, same with the CloudFormation bug explanation in the JSDoc. It's a bit confusing and shouldn't affect them. Despite the workaround, in effect we still have an unset value and the instance is selected according to the other criteria

Reply:
Thanks. Your PR had an excellent comment, so I used it as it was.

```ts
    // Despite what the CloudFormation schema says, the numeric properties are not optional.
    // An undefined value will cause the CloudFormation deployment to fail, e.g.
    // > Cannot invoke "java.lang.Integer.intValue()" because the return value of "software.amazon.codebuild.fleet.ComputeConfiguration.getMemory()" is null
```

Reply:
You're right. I also borrowed the description from your PR.